app/Nova/User.php

```diff
@@ -809,4 +813,36 @@ private static function adminCanRun(NovaRequest $request): bool
     {
         return $request->user()->hasRole('admin');
     }
+
+    /**
+     * Helper method: Graduation semester is a 6-digit code by default.
+     * Prepares semester data for display, returning empty if no string/invalid
+     * string is entered.
+     */
+    private static function parseGradSemester(?string $sem): string
```

Comment:
This code might be better placed as an attribute on the user model? as in, add a method called `getHumanReadableGraduationSemester` (unsure what a good name is, but the attribute value would be `Fall 2023`, etc. like you do here)

here's an example: https://github.com/RoboJackets/apiary/blob/b6b74b15383f419e9fed6de876bb6da4a357ca7d/app/Models/User.php#L397-L402

Reply:
I think that example is considered legacy now; [the Laravel docs have a different syntax](https://laravel.com/docs/10.x/eloquent-mutators#accessors-and-mutators). Same idea though.